### PR TITLE
Improve app start integration

### LINKED ISSUE #2103: pendingTimer when testing with SentryNavigatorObserver on sentry_flutter 8.2.0
### Platform

Flutter Mobile

### Obfuscation

Enabled

### Debug Info

Enabled

### Doctor

```txt
[✓] Flutter (Channel stable, 3.22.1, on macOS 14.5 23F79 darwin-arm64, locale en-GB)
    • Flutter version 3.22.1 on channel stable at /Users/alestiago/Developer/flutter/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a14f74ff3a (4 weeks ago), 2024-05-22 11:08:21 -0500
    • Engine revision 55eae6864b
    • Dart version 3.4.1
    • DevTools version 2.34.3

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/alestiago/Library/Android/sdk
    • Platform android-34, build-tools 33.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C500b
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Ultimate Edition (version 2023.1.4)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Community Edition (version 2023.1.4)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.89.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.90.0

[✓] Connected device (3 available)
    • macOS (desktop)                 • macos                 • darwin-arm64   • macOS 14.5
      23F79 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad • darwin         • macOS 14.5
      23F79 darwin-arm64
    • Chrome (web)                    • chrome                • web-javascript • Google Chrome
      125.0.6422.176

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

### Version

8.2.0

### Steps to Reproduce

1. Create a Flutter Application:
```sh
 flutter create my_app --platforms ios,android
```

2. Add `package:sentry_flutter`  8.2.0 as a dependency:
```sh
flutter pub add sentry_flutter:8.2.0
```

3. Update `lib/main.dart` to initialise Sentry:
```dart
Future<void> main() async {
  runZonedGuarded(() async {
    await SentryFlutter.init(
      (options) {
        options.dsn = 'https://example@sentry.io/add-your-dsn-here';
      },
    );

    runApp(const MyApp());
  }, (exception, stackTrace) async {
    await Sentry.captureException(exception, stackTrace: stackTrace);
  });
}
```

4. Update the `_incrementCounter` function to navigate somewhere:
```dart
 void _incrementCounter() {
    Navigator.push(context, MaterialPageRoute(builder: (context) {
      return const Text('Hello Sentry');
    }));
  }
```

5. Update the test in `widget_test` to navigate and assert for the text:

```dart
 testWidgets('Counter increments smoke test', (WidgetTester tester) async {
    await tester.pumpWidget(const MyApp());

    await tester.tap(find.byIcon(Icons.add));
    await tester.pumpAndSettle();

    expect(find.text('Hello Sentry'), findsWidgets);
  });
```

6. Run the test, the test should succeed.
```sh
fluter test
```

7. Add a `SentryNavigatorObserver` to the `MaterialApp` in `main.dart`:
```dart
MaterialApp(
  navigatorObservers: [
    SentryNavigatorObserver(),
  ],
  // other parameters
```

8. Run the test again, the test now fails.
```sh
fluter test
```

The test fails due to:
```txt
The following assertion was thrown running a test:
A Timer is still pending even after the widget tree was disposed.
'package:flutter_test/src/binding.dart':
Failed assertion: line 1527 pos 12: '!timersPending'
```

9. Change the `package:sentry_flutter` version from 8.2.0 to 8.1.0:
```sh
flutter pub add sentry_flutter:8.1.0
```

10. Run the test again, now the test succeeds.
```sh
fluter test
```

### Expected Result

The test succeeds in `package:sentry_flutter` v8.1.0, it should also succeed in v8.2.0.

### Actual Result

A test fails in `package:sentry_flutter` v8.2.0, but not in v8.1.0.

### Are you willing to submit a PR?

None

### LINKED ISSUE #2233: Transactions are very slow to complete when started inside `appRunner`, block the UI from showing up
### Platform

Flutter Mobile

### Obfuscation

Disabled

### Debug Info

Disabled

### Doctor

<details>
<summary>`flutter doctor -v`</summary>

```
[✓] Flutter (Channel stable, 3.22.3, on macOS 14.6 23G80 darwin-arm64, locale en-GB)
    • Flutter version 3.22.3 on channel stable at /Users/barabas/src/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b0850beeb2 (4 weeks ago), 2024-07-16 21:43:41 -0700
    • Engine revision 235db911ba
    • Dart version 3.4.4
    • DevTools version 2.34.3

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at /Users/barabas/Library/Android/sdk
    • Platform android-35, build-tools 35.0.0
    • ANDROID_HOME = /Users/barabas/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15F31d
    • CocoaPods version 1.15.2

[✗] Chrome - develop for the web (Cannot find Chrome executable at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)

[✓] Connected device (4 available)
    • Pixel 5 (mobile)                • 0C221FDD4003LH            • android-arm64 • Android 13 (API 33)
    • iPhone XR (mobile)              • 00008020-001E2C881A33002E • ios           • iOS 16.5.1 20F75
    • macOS (desktop)                 • macos                     • darwin-arm64  • macOS 14.6 23G80 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad     • darwin        • macOS 14.6 23G80 darwin-arm64

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

### Version

8.7.0

### Steps to Reproduce

Run this app on iOS or android in profile mode (using a real DSN):

```
Future<void> main() async {
  await SentryFlutter.init(
    (options) {
      options
        ..dsn =
            '<snip>'
        ..tracesSampleRate = 1.0;
    },
    appRunner: appMain,
  );
}

Future<void> appMain() async {
  final transaction = Sentry.startTransaction('hello', 'hello');
  await transaction.finish();

  runApp(MyApp());
}

```

### Expected Result

App UI appears almost instantly after app is launched.

### Actual Result

The UI appears after about 10s.

Rerun without the `tracesSamplerate` or without the `dsn` or without the `transaction` or with Sentry before and notice that the UI appears in less than a second. Removing just one of these is enough to avoid the bug.

I bisected on android and the first slow commit is https://github.com/getsentry/sentry-dart/commit/5e7abc5157b9e43973db9bc8e2a883d69d3fe136, where the app UI never appears.

### Are you willing to submit a PR?

No

### LINKED ISSUE #2259: Improve app start integration
### Description

Currently we use completers to wait for multiple things to finish before proceeding with the app start, this is problematic since it might block the event processing if for some reason the completer never finishes.

Things to change
- move app start ttid from navigator observer to app start integration (so root ttids are ignored in observer)
- thus app start integration is now the source of truth where app start happen
- handle case where `addPostFrameCallback` within the integration might not return, for example if `SentryFlutter.init` is called within an isolate
- deprecate `setAppStartEnd`

Fixes: 
- https://github.com/getsentry/sentry-dart/issues/2233
- https://github.com/getsentry/sentry-dart/issues/2103

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

- Always create root TTID transaction/span in app start integration
- Move app start processor functionality to integration

## :bulb: Motivation and Context

Closes #2259
Closes #2103
Closes #2233

## :green_heart: How did you test it?

- Unit tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: flutter/lib/src/integrations/native_app_start_integration.dart [hunk 1]
```diff
@@ -1,256 +1,29 @@
...
-            screenName, SentrySpanOperations.uiLoad,
-            startTimestamp: appStartInfo.start);
-        final ttidSpan = transaction.startChild(
-            SentrySpanOperations.uiTimeToInitialDisplay,
-            description: '$screenName initial display',
-            startTimestamp: appStartInfo.start);
-        await ttidSpan.finish(endTimestamp: appStartInfo.end);
-        await transaction.finish(endTimestamp: appStartInfo.end);
-      }
+      await _nativeAppStartHandler.call(hub, options, appStartEnd: appStartEnd);
```
Comment (buenaflor) [inline flutter/lib/src/integrations/native_app_start_integration.dart]: can we add a try/catch, to be more on the cautious side

File: flutter/lib/src/integrations/native_app_start_integration.dart [hunk 2]
```diff
@@ -1,256 +1,29 @@
...
+      this._frameCallbackHandler, this._nativeAppStartHandler);
 
-  final SentryNativeBinding _native;
   final FrameCallbackHandler _frameCallbackHandler;
-  final Hub _hub;
+  final NativeAppStartHandler _nativeAppStartHandler;
 
   /// This timestamp marks the end of app startup. Either set automatically when
   /// [SentryFlutterOptions.autoAppStart] is true, or by calling
   /// [SentryFlutter.setAppStartEnd]
```
Comment (buenaflor) [inline flutter/lib/src/integrations/native_app_start_integration.dart]: I'm not sure if `setAppStartEnd` now makes sense anymore. what if the user decides to trigger setAppStartEnd after we have already sent the transaction then it has no effect

essentially it's only usable when the user decides the app start end is before our app start integration triggers it?

File: flutter/lib/src/integrations/native_app_start_handler.dart [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+      return;
+    }
+    final appStartInfo = _infoNativeAppStart(nativeAppStart, appStartEnd);
+    if (appStartInfo == null) {
+      return;
+    }
+
+    // Create Transaction & Span
+
+    const screenName = SentryNavigatorObserver.rootScreenName;
```
Comment (buenaflor) [inline flutter/lib/src/integrations/native_app_start_handler.dart]: Can we move the `rootScreenName` somewhere else? I was confused at first because I thought we had a dependency to navigator observer here

File: flutter/lib/src/integrations/native_app_start_integration.dart [hunk 4]
```diff
@@ -1,256 +1,29 @@
...
-      );
-      setAppStartInfo(appStartInfo);
-      return;
-    }
-
-    if (_didFetchAppStart) {
-      return;
-    }
-
     _frameCallbackHandler.addPostFrameCallback((timeStamp) async {
```
Comment (buenaflor) [inline flutter/lib/src/integrations/native_app_start_integration.dart:34]: so currently if postFrameCallback is not triggered (for some reason) we just don't send app starts right?

File: flutter/lib/src/integrations/native_app_start_handler.dart [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+  String get appStartTypeOperation => 'app.start.${type.name}';
+
+  String get appStartTypeDescription =>
+      type == AppStartType.cold ? 'Cold Start' : 'Warm Start';
+  final pluginRegistrationDescription = 'App start to plugin registration';
+  final sentrySetupDescription = 'Before Sentry Init Setup';
+  final firstFrameRenderDescription = 'First frame render';
+}
+
+class TimeSpan {
```
Comment (buenaflor) [inline flutter/lib/src/integrations/native_app_start_handler.dart]:
```suggestion
class _TimeSpan {
```

can be private if I'm not wrong

File: flutter/lib/src/integrations/native_app_start_handler.dart [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+    await transaction.finish(endTimestamp: appStartInfo.end);
+  }
+
+  AppStartInfo? _infoNativeAppStart(
+    NativeAppStart nativeAppStart,
+    DateTime? appStartEnd,
+  ) {
+    final sentrySetupStartDateTime = SentryFlutter.sentrySetupStartTime;
+    if (sentrySetupStartDateTime == null) {
+      return null;
```
Comment (buenaflor) [inline flutter/lib/src/integrations/native_app_start_handler.dart:85]: adding this as a reminder that we probably don't need to return early

in case it is really null we only lose the sentryStartup span and firstFrameRender span

let's do that in a separate PR

File: flutter/lib/src/navigation/sentry_navigator_observer.dart [hunk 7]
```diff
@@ -365,12 +355,11 @@ class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
         return;
       }
 
-      if (isAppStart && endTimestamp != null) {
-        await _timeToDisplayTracker?.trackAppStartTTD(transaction,
```
Comment (buenaflor) [inline flutter/lib/src/navigation/sentry_navigator_observer.dart:372]: can we also remove `trackAppStartTTD` from the `TimeToDisplayTracker` class and from `TimeToInitialDisplayTracker`?

might also make sense to make a comment in `TimeToDisplayTracker` that app starts are managed solely by the app start integration

File: flutter/lib/src/integrations/native_app_start_integration.dart (hunk 2 above)
Comment (buenaflor) [reply flutter/lib/src/integrations/native_app_start_integration.dart]: we probably should deprecate it directly after this PR, this concept is a sentry flutter exclusive and I'm all for reducing complexity

Comment (buenaflor) [reply flutter/lib/src/integrations/native_app_start_integration.dart]: as mentioned in the sync, we'll deprecate it after we've ensured that the user has a proper alternative to do this

so we have to be careful to keep the same behaviour

File: flutter/lib/src/integrations/native_app_start_handler.dart (hunk 3 above)
Comment (denrase) [reply flutter/lib/src/integrations/native_app_start_handler.dart]: Not sure whats the best place, any suggestion?

File: flutter/lib/src/integrations/native_app_start_integration.dart (hunk 4 above)
Comment (denrase) [reply flutter/lib/src/integrations/native_app_start_integration.dart:34]: We pretty much have to assume it will be called, as we don't have a way to check this, except maybe using a completer again, but then we might block app start again like before.

File: flutter/lib/src/integrations/native_app_start_integration.dart [hunk 8]
```diff
@@ -1,256 +1,79 @@
...
-  final DateTime pluginRegistration;
-  final DateTime sentrySetupStart;
-
-  Duration? get duration => end?.difference(start);
-
-  SentryMeasurement? toMeasurement() {
-    final duration = this.duration;
-    if (duration == null) {
-      return null;
+  Future<void> callThrowing(
```
Comment (buenaflor) [inline flutter/lib/src/integrations/native_app_start_integration.dart]:
```suggestion
  Future<void> _callThrowing(
```

File: flutter/lib/src/integrations/native_app_start_integration.dart [hunk 9]
```diff
@@ -1,256 +1,79 @@
...
-      setAppStartInfo(appStartInfo);
-      return;
-    }
-
-    if (_didFetchAppStart) {
-      return;
+  void call(Hub hub, SentryFlutterOptions options) async {
+    if (!options.autoAppStart && _appStartEnd == null) {
+      await callThrowing(options, () async {
+        await _appStartEndCompleter.future.timeout(const Duration(seconds: 10));
```
Comment (buenaflor) [inline flutter/lib/src/integrations/native_app_start_integration.dart]: can you check if this is still an issue: https://github.com/getsentry/sentry-dart/issues/2103

File: flutter/lib/src/integrations/native_app_start_handler.dart (hunk 3 above)
Comment (buenaflor) [reply flutter/lib/src/integrations/native_app_start_handler.dart:39]: phew, tough call, could even be in the handler if we don't need it anywhere else` _rootScreenName`

Comment (buenaflor) [reply flutter/lib/src/integrations/native_app_start_handler.dart]: https://github.com/getsentry/sentry-dart/blob/feat/improve-app-start-integration/flutter/lib/src/navigation/sentry_navigator_observer.dart#L250C1-L253C1

we use it here but I think we can also remove this since we won't ever do a transaction on the root screen through the navigator observer since app starts/ttid is now managed through the integration

so we should be able to keep it in the app start handler as _rootScreenName

File: flutter/lib/src/integrations/native_app_start_integration.dart (hunk 9 above)
Comment (denrase) [reply flutter/lib/src/integrations/native_app_start_integration.dart]: had to move the completer into the post frame callback, otherwise we would block the ui again for the timeout duration, but it works now, also with #2103

File: flutter/lib/src/integrations/native_app_start_handler.dart (hunk 3 above)
Comment (denrase) [reply flutter/lib/src/integrations/native_app_start_handler.dart]: Do we really want to remove the root screen transactions from the observer? Maybe there is still value in having this, as the other is the whole app start, and here "only" of the root screen, no? We'd also be changing behaviour.

File: flutter/lib/src/integrations/native_app_start_integration.dart (hunk 4 above)
Comment (buenaflor) [reply flutter/lib/src/integrations/native_app_start_integration.dart:34]: that's fine, we will just go ahead with the statement that we need frame callbacks for app starts, otherwise we just don't report appstarts

File: flutter/lib/src/integrations/native_app_start_integration.dart [hunk 10]
```diff
@@ -1,256 +1,57 @@
...
-    if (_appStartCompleter.isCompleted) {
-      _appStartCompleter = Completer<AppStartInfo?>();
-    }
-    _appStartCompleter.complete(appStartInfo);
-  }
+  final NativeAppStartHandler _nativeAppStartHandler;
+  DateTime? _appStartEnd;
 
+  /// This timestamp marks the end of app startup. Either set by calling
+  // ignore: deprecated_member_use_from_same_package
```
Comment (buenaflor) [inline flutter/lib/src/integrations/native_app_start_integration.dart]:
```suggestion
```

let's deprecate it only when we've ensured that the user has an alternative.

File: flutter/lib/src/integrations/native_app_start_integration.dart [hunk 11]
```diff
@@ -1,256 +1,57 @@
...
-        // If the system forked the process earlier to accelerate the app start.
-        // And some unknown reasons that could not be reproduced.
-        // We've seen app starts with hours, days and even months.
-        if (duration != null && duration.inMilliseconds > _maxAppStartMillis) {
-          setAppStartInfo(null);
-          return;
+      try {
+        if (!options.autoAppStart && _appStartEnd == null) {
+          await _appStartEndCompleter.future
+              .timeout(const Duration(seconds: 10));
```
Comment (buenaflor) [inline flutter/lib/src/integrations/native_app_start_integration.dart:38]: this means that users who want to manually set an app start cannot do so anymore if the postFrameCallback is not triggered for example if they initialize the sdk in an isolate.

I fear that this might be a breaking change for those users who rely on that, are there any alternatives we can do for those?

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Unreleased
+
+### Enhancements
+
+- Improve app start integration ([#2266](https://github.com/getsentry/sentry-dart/pull/2266))
```
Comment (buenaflor) [inline CHANGELOG.md:7]: let's mention that this fixes blocking issues such as https://github.com/getsentry/sentry-dart/issues/2233

File: flutter/lib/src/sentry_flutter_options.dart [hunk 13]
```diff
@@ -176,6 +176,7 @@ class SentryFlutterOptions extends SentryOptions {
   /// Automatically track app start measurement and send it with the
   /// first transaction. Set to false when configuring option to disable or if
   /// you want to set the end time of app startup manually using
+  // ignore: deprecated_member_use_from_same_package
```
Comment (buenaflor) [inline flutter/lib/src/sentry_flutter_options.dart]:
```suggestion
```

let's deprecate it only when we've ensured that the user has an alternative.

File: flutter/lib/src/integrations/native_app_start_integration.dart (hunk 11 above)
Comment (buenaflor) [reply flutter/lib/src/integrations/native_app_start_integration.dart:38]: I'm not sure here if it's fine to release this in `8.10.0` and add it to the changelog that this is a breaking change for app start under special circumstances

Comment (buenaflor) [reply flutter/lib/src/integrations/native_app_start_integration.dart:38]: on the other hand the amount of users that do this should be minimal. 

cc @stefanosiano any thoughts on this?

Comment (denrase) [reply flutter/lib/src/integrations/native_app_start_integration.dart:38]: Having this outside of `postFrameCallback` blocks the UI when awaiting the completer, so users are never able to even call `setAppStartEnd`.

What we could do is not call the it if we are not in the main isolate. wdyt?

Comment (buenaflor) [reply flutter/lib/src/integrations/native_app_start_integration.dart:38]: is it possible for it to trigger when not on a main isolate?

and would that still fix this issue https://github.com/getsentry/sentry-dart/issues/2103

Comment (buenaflor) [reply flutter/lib/src/integrations/native_app_start_integration.dart:38]: generally I don't mind keeping the impl like it is now if we cant find alternatives, I'm only unsure whether it's fine to just mark it as breaking change for those situations in a minor release

Comment (denrase) [reply flutter/lib/src/integrations/native_app_start_integration.dart:38]: So I added a test, and it would work, but we have an issue we had before. While we can use `Isolate.current.debugName` to check for `main` isolate, nothing in the documentation indicates that this is guaranteed to work.

Maybe we should document that, for now, this integration only works with sentry initialised on the main isolate. 

Technically it would not be breaking, as the previous implementation also relied upon the postFrameCallback and was broken anyway. We ran in the completer timeout and users could not manually set app start end anyway.

Also, whats the point of doing a ui transaction if we are not on the main isolate? All UI work is done on main, so per definition there should be no ui transactions on other isolates, no? 🤔

Comment (buenaflor) [reply flutter/lib/src/integrations/native_app_start_integration.dart:38]: Let's keep app starts for main isolate and it would just not be reported if the SDK was initialized in another isolate.

**Comment by buenaflor** [comment]
@denrase can you fix the last test pls

File: flutter/lib/src/sentry_flutter.dart [hunk 14]
```diff
@@ -225,9 +228,15 @@ mixin SentryFlutter {
...
   /// Manually set when your app finished startup. Make sure to set
-  /// [SentryFlutterOptions.autoAppStart] to false on init.
+  /// [SentryFlutterOptions.autoAppStart] to false on init. The timeout duration
+  /// for this to work is 10 seconds.
   static void setAppStartEnd(DateTime appStartEnd) {
-    NativeAppStartIntegration.appStartEnd = appStartEnd;
+    // ignore: invalid_use_of_internal_member
+    final integrations = Sentry.currentHub.options.integrations
+        .whereType<NativeAppStartIntegration>();
+    for (var integration in integrations) {
```
Comment (buenaflor) [inline flutter/lib/src/sentry_flutter.dart]:
```suggestion
    for (final integration in integrations) {
```

**Review by buenaflor — APPROVED** [review]
nice!

only got one last comment

**Comment by denrase** [comment]
We have some 403 errors in CI. I'm gonna go ahead and merge, since they are not related to this PR.

```
Error: Exception in thread "main" org.kohsuke.github.HttpException: {"message":"Resource not accessible by integration","documentation_url":"https://docs.github.com/rest","status":"403"}
```
